connection.go

```diff
@@ -238,6 +243,22 @@ func getTimeout(ctx context.Context) time.Duration {
 	return deadline.Sub(time.Now())
 }
 
+func (co ConnectionOptions) withDefaults() ConnectionOptions {
+	if co.ChecksumType == ChecksumTypeNone {
+		co.ChecksumType = ChecksumTypeCrc32
+	}
+	if co.FramePool == nil {
+		co.FramePool = DefaultFramePool
+	}
+	if co.SendBufferSize <= 0 {
+		co.SendBufferSize = defaultConnectionBufferSize
+	}
+	if co.RecvBufferSize <= 0 {
+		co.RecvBufferSize = defaultConnectionBufferSize
```

Comment:
Is it actually useful to distinguish between Recv and Send buffer sizes?  Will they ever actually be different?